iterreg/sparse/solvers.py

```diff
@@ -253,4 +260,49 @@ def fista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=10,
             all_w[t // f_store] = w
             if verbose:
                 print(t, E[t // f_store])
+
+            if (t > 0) and (np.abs(E[t // f_store - 1]-E[t // f_store]) <
+                            tol * E[0]):
+                return w, all_w[:t//f_store + 1], E[:t//f_store + 1]
     return w, all_w, E
+
+
+@njit
+def reweighted(X, y, alpha, deriv, max_iter=1_000, n_adapt=5,
+               f_store=10, w_init=None, tol=0., verbose=False):
+    """Reweighted for the Tikhonov problem."""
+    p = X.shape[1]
+    X = np.asfortranarray(X)
+    lc = np.zeros(p)
```

Comment:
Numba doesn't work with 'axis' term

Reply:
you can jit only the cd loop isolated in a function, this way you 
- reduce the compilation time (this can be important)
- keep the full numpy flexibility elsewhere

Reply:
but this is cosmetic. keep it this way, I did not notice that the full function was jitted. I think ord=1 will not work either in that case, so revert to what you had correctly written before